force-app/main/default/classes/CRLP_RollupUI_SVC.cls

```diff
@@ -173,6 +182,8 @@ public with sharing class CRLP_RollupUI_SVC {
     */
     @AuraEnabled
     public static String setupRollupDetail(String rollupId, List<String> targetObjectNames, List<String> detailObjectNames) {
+        if (!STG_Panel.runningUserIsAdmin()) throwAuraPermsException();
```

Comment:
Style-wise, I think we prefer to have if statements w/ curly brackets. I know that'd expand these statements out to three lines from one, and there's a small handful of them in this file. What if we moved this check & throw into a function of its own?

```suggestion
        throwIfRunningUserNotAdmin();
```

```public static void throwIfRunningUserNotAdmin() {
    if(!STG_Panel.runningUserIsAdmin()) {
        throwAuraPermsException();
    }
}```

Reply:
Great suggestion @bdvorachek. I wonder if we should consider putting that method in the `STG_PanelUserRollup_CTRL` or `STG_SettingsService` class?

Reply:
@dfuller114 Or maybe even the `UTIL_Permissions` class?

Reply:
@bdvorachek @force2b I like the idea of having this be its own method.
I would like this to not be tied to NPSP Settings since this level of `AuraEnabled` security is not related to NPSP Settings UI. I'm also not sure if this would fit in `UTIL_Permissions` since that class is intended for Object CRUD and FLS checking and this is more API focused permission checking with a client. What about having it in `USER_UserService` where we determine if a user is an admin in the first place?

I can see the user service being refactored in the future to handle entities of different user types such as an `AdminUser` entity that is constructed by `CRLP_RollupUI_SVC` at the top of each of these methods and if that constructor fails because the user is not an admin, the constructor itself could throw the error or something like that. All that to say, it seems to make since to push this logic down as far as we can near where an admin user is determined to allow for refactors like that in the future.

Thoughts?

Reply:
@dfuller114 In this case, the security might be specific to NPSP Settings since we're calling the `runningUserIsAdmin()` method. (CRLP Configuration is accessible through NPSP settings, even though the classes don't start with STG). So, maybe using STG_SettingsService or SETG_PanelUserController could work. Other implementations of this kind of security check, on other pages, might check for different perms and possibly through a different exception.

Reply:
`STG_PanelUserRollup_CTRL` makes sense since it is already performing other user permissions operations related to the user's interaction with configuring rollups. Just pushed up those changes.